### PR TITLE
[WIP] Ports Virgo's 'Busy Space'

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request
Ports Virgo's "Busy Space", airtraffic announcements from the local system broadcasted right to your station!
Obviously, modified for our lore, and tweaked to work better here in general.

It can be toggled via the Automated Announcement System in the telecomms room; this WONT toggle docking/departure announcements from the ERT Ferry, Arrivals Relay, or Evac Shuttle
<hr>

**TO-DO:**
All of it

- [ ] Not be horrible Hyper/Virgo code
- [ ] Start on roundstart, not server load
- [ ] Announce for ERT Ferry; toggle-able onboard the Ferry (for non-certified ERTs)
- [ ] Announce for Arrivals Relay
- [ ] Relay leaves for repairs if damaged, announcement for such (TG shuttle already leaves for repairs, ours should as well)
- [ ] Announce for Evac Shuttle
- [x] Working toggle (maybe toggleable from announcement computer AND Automated Announcement System?)
- [x] Let admins place their own atc announcements?
- [ ] Fill in our own lore

## How This Contributes To The Skyrat Roleplay Experience
Seeing lore via in-game means is good.
The universe feeling alive is good.
Hearing when the damn Relay is docked at the station is good because I'm sick of missing my flight.

## Changelog

:cl:
This will be filled out when the PR has actual progress
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/busy_space/loremaster.dm [hunk 1]
```diff
@@ -0,0 +1,11 @@
+var/datum/lore/loremaster/loremaster = new/datum/lore/loremaster
+
+/datum/lore/loremaster
```
Comment (axietheaxolotl) [inline modular_skyrat/modules/busy_space/loremaster.dm:3]: wtf we have lore

Comment (ghost) [reply modular_skyrat/modules/busy_space/loremaster.dm:3]: here it comes baby 😎 

Most of it'll be kept in organizations.dm

**Comment by YakumoChen** [comment]
Lore wise I'm not really sure "busy" space works here. The station is set in Frontier space, which would be typically the outside of inhabited space or major trade routes. 

What kind of air traffic is likely to come out here aside from pirates or other types trying to _avoid_ air traffic?

Less to the point, the server is very busy and having more pointless stuff filling up the radio does not really sound like a good time. There are quite a few jobs that have to sort through all the useless radio chatter to get important info (namely security)

**Comment by ghost** [comment]
> Lore wise I'm not really sure "busy" space works here. The station is set in Frontier space, which would be typically the outside of inhabited space or major trade routes.
> 
> What kind of air traffic is likely to come out here aside from pirates or other types trying to _avoid_ air traffic?
> 
> Less to the point, the server is very busy and having more pointless stuff filling up the radio does not really sound like a good time. There are quite a few jobs that have to sort through all the useless radio chatter to get important info (namely security)

"Frontier" is a vague and incomplete term. To be more specific, this is the _forefront_ of the frontier, the boomtown out into the unknown. There'll be lots of traffic going to new projects and other nearby areas. Lore-wise, it works fine; it happens once every 20ish minutes usually, so it's really not BUSY or comm-spammy.
Moreover, the main hope of this is to have a system for events and nearspace announcements, such as ERT docking and the arrivals relay's current position. Even at that point, the ERT ship can toggle their announcement from their shuttle, and the AAS can disable the random ones if the station is too busy; I may even have it automatically disable on red alert or higher, tbh

**Comment by ghost** [comment]
eh

**Comment by Guidesu** [comment]
Don't give up! Its an mere suggestion but you can scale down things. I would love it on game and i can even help you!

**Comment by ghost** [comment]
> Don't give up! Its an mere suggestion but you can scale down things. I would love it on game and i can even help you!

I'll come back to it eventually once more shit is in order, dont worry too much. Just, not motivated for something like this rn.

**Pull request closed without merging**